packages/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/SharedPreferencesPlugin.java

```diff
@@ -5,199 +5,36 @@
-      stream = new ObjectOutputStream(byteStream);
-      stream.writeObject(list);
-      stream.flush();
-      return Base64.encodeToString(byteStream.toByteArray(), 0);
-    } finally {
-      if (stream != null) {
-        stream.close();
-      }
-    }
-  }
-
-  // Filter preferences to only those set by the flutter app.
-  private Map<String, Object> getAllPrefs() throws IOException {
-    Map<String, ?> allPrefs = preferences.getAll();
-    Map<String, Object> filteredPrefs = new HashMap<>();
-    for (String key : allPrefs.keySet()) {
-      if (key.startsWith("flutter.")) {
-        Object value = allPrefs.get(key);
-        if (value instanceof String) {
-          String stringValue = (String) value;
-          if (stringValue.startsWith(LIST_IDENTIFIER)) {
-            value = decodeList(stringValue.substring(LIST_IDENTIFIER.length()));
-          } else if (stringValue.startsWith(BIG_INTEGER_PREFIX)) {
-            String encoded = stringValue.substring(BIG_INTEGER_PREFIX.length());
-            value = new BigInteger(encoded, Character.MAX_RADIX);
-          } else if (stringValue.startsWith(DOUBLE_PREFIX)) {
-            String doubleStr = stringValue.substring(DOUBLE_PREFIX.length());
-            value = Double.valueOf(doubleStr);
-          }
-        } else if (value instanceof Set) {
-          // This only happens for previous usage of setStringSet. The app expects a list.
-          List<String> listValue = new ArrayList<>((Set) value);
-          // Let's migrate the value too while we are at it.
-          boolean success =
-              preferences
-                  .edit()
-                  .remove(key)
-                  .putString(key, LIST_IDENTIFIER + encodeList(listValue))
-                  .commit();
-          if (!success) {
-            // If we are unable to migrate the existing preferences, it means we potentially lost them.
-            // In this case, an error from getAllPrefs() is appropriate since it will alert the app during plugin initialization.
-            throw new IOException("Could not migrate set to list");
-          }
-          value = listValue;
-        }
-        filteredPrefs.put(key, value);
-      }
-    }
-    return filteredPrefs;
+  @Override
+  public void onAttachedToEngine(@NonNull FlutterPlugin.FlutterPluginBinding binding) {
+    setupChannel(binding.getFlutterEngine().getDartExecutor(), binding.getApplicationContext());
   }
 
-  private void commitAsync(final Editor editor, final MethodChannel.Result result) {
-    new AsyncTask<Void, Void, Boolean>() {
-      @Override
-      protected Boolean doInBackground(Void... voids) {
-        return editor.commit();
-      }
-
-      @Override
-      protected void onPostExecute(Boolean value) {
-        result.success(value);
-      }
-    }.execute();
+  @Override
+  public void onDetachedFromEngine(@NonNull FlutterPlugin.FlutterPluginBinding binding) {
+    channel.setMethodCallHandler(null);
```

Comment:
Recommendation: Consider defining a `teardownChannel()` for symmetry with `setupChannel()`.